homeassistant/components/vesync/switch.py

```diff
@@ -45,53 +73,45 @@ def _setup_entities(
     async_add_entities,
     coordinator: VeSyncDataCoordinator,
 ):
-    """Check if device is a switch and add entity."""
-    entities: list[VeSyncBaseSwitch] = []
-    for dev in devices:
-        if DEV_TYPE_TO_HA.get(dev.device_type) == "outlet":
-            entities.append(VeSyncSwitchHA(dev, coordinator))
-        elif DEV_TYPE_TO_HA.get(dev.device_type) == "switch":
-            entities.append(VeSyncLightSwitch(dev, coordinator))
-
-    async_add_entities(entities, update_before_add=True)
-
+    """Check if device is online and add entity."""
+    async_add_entities(
+        (
+            VeSyncSwitchEntity(dev, description, coordinator)
+            for dev in devices
+            for description in SENSOR_DESCRIPTIONS
+            if description.exists_fn(dev)
+        ),
+    )
 
-class VeSyncBaseSwitch(VeSyncBaseEntity, SwitchEntity):
-    """Base class for VeSync switch Device Representations."""
 
-    _attr_name = None
+class VeSyncSwitchEntity(SwitchEntity, VeSyncBaseEntity):
+    """VeSync switch entity class."""
 
-    def turn_on(self, **kwargs: Any) -> None:
-        """Turn the device on."""
-        self.device.turn_on()
+    def __init__(
+        self,
+        device: VeSyncBaseDevice,
+        description: VeSyncSwitchEntityDescription,
+        coordinator: VeSyncDataCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(device, coordinator)
+        self.entity_description: VeSyncSwitchEntityDescription = description
```

Comment:
overwrite the type outside of the constructor instead of here, it usually looks a bit cleaner

Reply:
Adjusted.